### PR TITLE
[feat] sbert 위키피디아 마스킹 실험 코드 생성 및 성능 개선

### PR DESCRIPTION
**Pull request by always-spring01**
## 개요
- 기존의 sbert 실험 함수는 **2중 for문**을 사용하여 계산하여 연산 시간이 매우 높은 단점이 있었음
- 이를 gpu 연산 처리를 통해 극복할 수 있었음 (177h -> 27s)
- 위키피디아 마스킹 데이터셋을 통으로 올리는 것보다 로컬에서 실험하도록 데이터셋을 제거하는 것이 옳다 판단하여 제거하였음
- 위키피디아 마스킹 데이터셋에 맞춰진 테스트 함수 및 테스트 코드 생성

## 작업 내용
- 일부 데이터셋 (위키피디아 마스킹 데이터셋)의 구조 및 업로드 방식을 변경
- sbert 전처리 함수를 별도의 `python` 파일로 정의
- sbert 인코딩 함수를 별도의 `python` 파일로 정의
- sbert 테스트 함수를 별도의 `python` 파일로 정의
- 최종 sbert 실험 테스트 파일을 `main.ipynb`로 정의

## 테스트 방법
- `main.ipynb` 파일을 실행해주시면 됩니다!
- **기존의 `main.ipynb` 파일은 `sample.ipynb` 파일로 이동하였습니다!**
- 추후 해당 `sample.ipynb` 파일은 삭제할 예정입니다.

## 리뷰 요청 사항
- 각자 `main.ipynb`를 돌려보시고, 개인이 생성한 데이터셋에서 실험이 가능한지 확인 부탁드립니다.
- 추가적으로 해당 코드를 실행하기 위해서는 강력한 컴퓨팅 자원이 필요합니다 -> **NVIDIA TITAN Xp** GPU (vram 12gb) 이상의 그래픽카드에서 torch, cuda를 설치한 이후 진행하시는 것을 권장합니다.

### CODE REVIEW

File: research/data/dataset_masking/.gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
-ko-wiki-masked
\ No newline at end of file
+ko-wiki-masked
+ko-wiki
```
Comment (JuYeobOh) [inline research/data/dataset_masking/.gitignore:2]: 데이터를 올리지 않는 것은 동의합니다. 하지만 기존 코드를 보니 랜덤 시드를 고정할 수 있어 보이지 않습니다. 그러면 사람마다 결과가 달라질 수 있지 않을까요?
실험 단계에선 각자 해도 괜찮다고 생각이 들지만, 평가 단계에선 서로 데이터를 맞출 방법이 필요할거 같습니다.

File: research/sentence-bert/check.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            with open(f"{encode_path1}/{path1}/{file1}", "rb") as f1:
+                pivot_data = np.array(pickle.load(f1))
+            pivot_tensor = torch.tensor(pivot_data, dtype=torch.float32, device="cuda")
+
+            for file2 in pkl_list2:
+                with open(f"{encode_path2}/{path1}/{file2}", "rb") as f2:
+                    compare_data = np.array(pickle.load(f2))
+                compare_tensor = torch.tensor(compare_data, dtype=torch.float32, device="cuda")
+
+                sim_matrix = torch.mm(pivot_tensor, compare_tensor.T) # GPU 가속 필요
```
Comment (JuYeobOh) [inline research/sentence-bert/check.py:37]: 벡터 내적을 통해서 연산 속도를 개선시킨거 같은데 맞나요? 좋은거 같습니다. 
근데, 질문이 하나 있습니다. 해당 코드는 단순 내적이라 코사인 유사도와 다른 것 같은데, 같은 threshold를 적용하는 것이 맞나요?

코사인 유사도 공식을 보았을 때 크기로 나눠주어야 한다고 생각했습니다.

**Review by JuYeobOh — COMMENTED** [review]
177시간을 27초로 개선 시킨 것은 엄청난 성과인 것 같습니다. 🥇 
저도 이후 코드에 적용하도록 하겠습니다. 

궁금한 사항에 대해 별도의 코드 리뷰로 남겼으니 확인 부탁드립니다.

File: research/sentence-bert/check.py (hunk 2 above)
Comment (always-spring01) [reply research/sentence-bert/check.py:37]: 코사인 유사도 공식의 분모 부분에 따른 질문으로 인지하였습니다.

해당 `_tensor` 데이터들은 `sbert`를 통해 `normalization`을 수행한 값들입니다. 정규화가 사전에 수행되었으므로 코사인 유사도의 분모 부분은 **1**이 되기 때문에, 해당 분모는 생략하여 속도를 더욱 높히고자 하였습니다.

Comment (JuYeobOh) [reply research/sentence-bert/check.py:37]: 감사합니다.

**Comment by JuYeobOh** [comment]
이상 없는 것 같습니다. 머지하도록 하겠습니다.
